cloudify_hostpool/config/yaml_pool.py

```diff
@@ -76,7 +77,11 @@ def _create_host(_host):
                     "'host' or the 'ip_range' key")
 
     def _get_auth(self, host):
-        return host.get('auth') or self.default.get('auth')
+        auth = host.get('auth') or self.default.get('auth', {})
+        if auth.get('keyfile') and not os.access(auth['keyfile'], os.R_OK):
+            raise exceptions.ConfigurationError(
+                'There is no key file: {0}'.format(auth['keyfile']))
```

Comment:
The message should reflect a permissions problem as well, corresponding to the conditions. 
maybe "keyfile {0} does not exist or does not have the proper permissions"